src/Glpi/Http/AccessErrorListener.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+/**
+ * ---------------------------------------------------------------------
+ *
+ * GLPI - Gestionnaire Libre de Parc Informatique
+ *
+ * http://glpi-project.org
+ *
+ * @copyright 2015-2024 Teclib' and contributors.
+ * @licence   https://www.gnu.org/licenses/gpl-3.0.html
+ *
+ * ---------------------------------------------------------------------
+ *
+ * LICENSE
+ *
+ * This file is part of GLPI.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * ---------------------------------------------------------------------
+ */
+
+namespace Glpi\Http;
+
+use Glpi\Exception\Access\AccessException;
+use Glpi\Exception\Access\RequiresHttpsException;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpKernel\Event\ExceptionEvent;
+use Symfony\Component\HttpKernel\KernelEvents;
+
+class AccessErrorListener implements EventSubscriberInterface
+{
+    public static function getSubscribedEvents(): array
+    {
+        return [
+            KernelEvents::EXCEPTION => 'onKernelException',
+        ];
+    }
+
+    public function onKernelException(ExceptionEvent $event): void
+    {
+        $e = $event->getThrowable();
+
+        if (!$e instanceof AccessException) {
+            return;
+        }
```

Comment:
Maybe we could have a `Glpi\Http\Exception\HttpExceptionInterface` that would describe the `setRequest()` and `asResponse()`. All the exceptions we want to handle here are indeed thrown in an HTTP request handling context.

We could then extend most of the `Symfony\Component\HttpKernel\Exception\HttpException` exceptions we would use in order to be able to adapt the response, when necessary.

Reply:
The problem is that interfacing exceptions is quite complicated because it will always end up with a specific class, abstract or not. Since PHP only has the `Throwable` for exceptions, you cannot make a class that "just implements it". You must either extend `Error` or `Exception`. Here I chose `RuntimeException` for the sole purpose that all exceptions regarding HTTP are explicitly called at runtime, not parse-time or compile-time.

Though the interface might seem interesting, it will be useless compared to just the `AccessException` abstract class.

I could rename it `HttpException` though, I think that would indeed make sense.

Reply:
We just have to provide a way to give the developpers a way to throw an exception that will be used, most of the time, to craft a response with a translated message and a specific HTTP response code.

For the specific responses (redirects when session does not exists, for instance), it could be a specific behaviour implemented in a lsiter or even in the error controller, but I do not think we have to build a system that can be extended by plugins.

Whatever the solution is, it just need to be easy to use and to maintain.

Reply:
> We just have to provide a way to give the developpers a way to throw an exception that will be used, most of the time, to craft a response with a translated message and a specific HTTP response code.

For that case, ANY instance of `Symfony\Component\HttpKernel\Exception\HttpException` will do. No need for GLPI to have its own.

> For the specific responses (redirects when session does not exists, for instance), it could be a specific behaviour implemented in a lsiter or even in the error controller, but I do not think we have to build a system that can be extended by plugins.

That's for future cases: ideally, these should happen even **before** any Controller is resolved.
For instance, the "redirect if no session" is exactly what Symfony's Security component is about in its "Authentication" section. There are tons of strategies (using SSO, or user/password, or API tokens, etc, IP-based rate limiting, etc.) that **must** be checked even before Session is initialized or checked. For now, handling it with simple Exception classes is really the most straightforward way to do it without breaking the app or bringing too much complexity with new concepts